### DISCUSSION TITLE
Upgrade ideas

### DISCUSSION THREAD

**Discussion by enriquespsa** [malliaridis/obssd#42] [Ideas]
+ Add the possibility to see the sources of another scene without changing the scene.

+ Add support for audio sources when they are grouped.

+ Add support for audio sources when they are added through a scene.

+ Add a record pause button.

+ Add the possibility to see the sources contained in a group.

+ Group the functions of the "Help, Developer's News and App Info" buttons into a single button.
---------------------------------------------------------------------------------------------------------------------------------------------------
💬 I can't think of any more ideas, the app is quite useful at the moment and those ideas would make it even better.

💬 I have used the Google translator, I hope they are understandable

**Comment by malliaridis** [comment]
Thanks for your suggestions. Let me go through the suggestions one by one:

> Add the possibility to see the sources of another scene without changing the scene.

This feature was something I considered adding to the app, but I dropped the idea because the accessibility into other scenes' sources would be hidden behind a long click action, which many users don't think of. But since there is at least one use case that could come in handy, I will consider adding that after the next major release.

> Add support for audio sources when they are grouped.

Groups / folders are indeed planned and partly implemented in the next release. When you say grouped audio sources, are you talking about sources / scene items, or are you talking about the mixers? Because mixers themselves are not groupable as far as I know.

> Add support for audio sources when they are added through a scene.

I think this is a limitation of the API I am using. I created a [bug report](https://github.com/malliaridis/obssd/issues/43) for tracking it's status and I will look into that before the next release is rolled out.

> Add a record pause button.

I planned to add one in the upcoming release, but the Streamlabs OBS API does not support pausing a recording, so the function would be limited to only OBS Studio. I might find a workaround or offer the option to pause only for OBS Studio, but for now there are no plans adding that in the near future. Feel free to create a new discussion for this one if it is important to you to have this option

> Add the possibility to see the sources contained in a group.

As mentioned before, grouping will be supported in the upcoming version and hopefully with at least the most significant usages. :)

> Group the functions of the "Help, Developer's News and App Info" buttons into a single button.

I have received a lot of feedback from people that have issues finding the PC's IP address. Those people were not able to use the app at all due to this issue. If they were using the help they would have at least an idea where to look for an IP address. So hiding the Help inside another button would probably make it even worse to find any help regarding this issue. I added those three buttons to the end together with the settings because they are less important and the user doesn't have to scroll for frequently used features.

May I ask therefore, what's the background of this idea?

And last but not least, thank you very much for your time and for writing those ideas down for me. I really appreciate that! 😄

**Reply by enriquespsa** [reply]
+ Yes, I am referring to sources/scene elements that when grouped are not detected in the app's mixer (Add support for audio sources when grouped.)

+ My intention when suggesting the ideas is to use the app with the essential functionalities for me, it seems to me a non-intrusive and simple app that, since it does not require server software on the PC, is more comfortable.

                                                        
> **Thanks for the answer and for the attention.**

**Reply by malliaridis** [reply]
I see, I think that is probably a limitation of the plugin. I read a message from one of the developers that said groups are very buggy in OBS Studio already, and that nested scenes should be preferred. But maybe with the next version it will work better. I will look into that in more detail once everything is ready for release. :)

Yeah, the app's goal is not to cover every functionality available on OBS Studio, but instead provide a remote stream deck for doing just the most important stuff while streaming. So I try to avoid anything that could make the app more complicated.
